File: GeneratedEntityReaderClasses/Components/AltarComponent.cs
namespace GeneratedNoitaClasses;
public class AltarComponent : NoitaComponentBase,  iNoitaType<AltarComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString recognized_entity_tags { get; set; } = new();
	public NoitaInt good_fx_material { get; set; } = new();
	public NoitaInt neutral_fx_material { get; set; } = new();
	public NoitaInt evil_fx_material { get; set; } = new();
	public NoitaInt uses_remaining { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		recognized_entity_tags.Read(s);
		good_fx_material.Read(s);
		neutral_fx_material.Read(s);
		evil_fx_material.Read(s);
		uses_remaining.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		recognized_entity_tags.Write(s);
		good_fx_material.Write(s);
		neutral_fx_material.Write(s);
		evil_fx_material.Write(s);
		uses_remaining.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/AltarComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class AltarComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    recognized_entity_tags:NoitaString = field(default_factory=NoitaString)
    good_fx_material:NoitaInt = field(default_factory=NoitaInt)
    neutral_fx_material:NoitaInt = field(default_factory=NoitaInt)
    evil_fx_material:NoitaInt = field(default_factory=NoitaInt)
    uses_remaining:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.recognized_entity_tags.Read(s)
        self.good_fx_material.Read(s)
        self.neutral_fx_material.Read(s)
        self.evil_fx_material.Read(s)
        self.uses_remaining.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.recognized_entity_tags.Write(s)
        self.good_fx_material.Write(s)
        self.neutral_fx_material.Write(s)
        self.evil_fx_material.Write(s)
        self.uses_remaining.Write(s)

